homeassistant/components/daikin/sensor.py

```diff
@@ -117,6 +124,11 @@ def state(self):
             return self._api.device.inside_temperature
         if self._device_attribute == ATTR_OUTSIDE_TEMPERATURE:
             return self._api.device.outside_temperature
+
+        if self._device_attribute == ATTR_HUMIDITY:
+            return int(self._api.device.values.get(ATTR_HUMIDITY).replace("-", "0"))
```

Comment:
This kind of logic should be handled by the pydaikin package
Please open a PR there https://bitbucket.org/mustang51/pydaikin/src/master/

Reply:
> Thank you for review, I'll check pydaikin and send PR there.

Looking forward for a pr to pydaikin!

Reply:
I have updated pydaikin but I am not able to open PR on bitbucket.. :-(

Here's patch:

```
From 6080004e98fc66418b397afb7915e9e2a7c5ef4a Mon Sep 17 00:00:00 2001
From: Filip Pytloun <filip@pytloun.cz>
Date: Fri, 5 Jun 2020 16:51:38 +0200
Subject: [PATCH] Add humidity sensor support

---
 pydaikin/daikin_base.py | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/pydaikin/daikin_base.py b/pydaikin/daikin_base.py
index 5a7aab6..40990da 100644
--- a/pydaikin/daikin_base.py
+++ b/pydaikin/daikin_base.py
@@ -312,6 +312,13 @@ class Appliance:  # pylint: disable=too-many-public-methods
         except (TypeError, ValueError):
             return None
 
+    def _humidity(self, dimension):
+        """Parse humidity."""
+        try:
+            return int(self.values.get(dimension).replace("-", "0"))
+        except (TypeError, ValueError):
+            return None
+
     @property
     def device_ip(self):
         """Return device's IP address."""
@@ -342,6 +349,11 @@ class Appliance:  # pylint: disable=too-many-public-methods
         """Return True if the device is not an AirBase unit."""
         return self.outside_temperature is not None
 
+    @property
+    def support_humidity(self):
+        """Return True if the device has humidity sensor."""
+        return self.humidity is not None
+
     @property
     def support_energy_consumption(self):
         """Return True if the device supports energy consumption monitoring."""
@@ -364,6 +376,16 @@ class Appliance:  # pylint: disable=too-many-public-methods
         """Return current target temperature."""
         return self._temperature('stemp')
 
+    @property
+    def humidity(self):
+        """Return current humidity."""
+        return self._humidity('hhum')
+
+    @property
+    def target_humidity(self):
+        """Return target humidity."""
+        return self._humidity('shum')
+
     def energy_consumption(self, mode=ATTR_TOTAL, time=TIME_TODAY):
         """Return today energy consumption in kWh."""
         parser = self.ENERGY_CONSUMPTION_PARSERS.get(f'{mode}_{time}')
-- 
2.25.1
```

Reply:
Here is the PR: https://bitbucket.org/mustang51/pydaikin/pull-requests/30/add-humidity-sensor-support